### PR TITLE
Add lua file system library (for CoC Compatibility)

### PR DESCRIPTION
**Pull request by ChugunovRoman**

### CODE REVIEW

File: Externals/libtheora_static.vcxproj [hunk 1]
```diff
@@ -95,4 +95,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
\ No newline at end of file
+</Project>
```
Comment (Xottab-DUTY) [inline Externals/libtheora_static.vcxproj]: A new line on the end of file was added, please, remove this change

File: src/xrScriptEngine/script_engine.cpp [hunk 2]
```diff
@@ -926,7 +927,7 @@ void CScriptEngine::init(ExporterFunc exporterFunc, bool loadGlobalNamespace)
     {
         const bool nilConversion =
             pSettingsOpenXRay->read_if_exists<bool>("lua_scripting", "allow_nil_conversion", true);
-     
+
```
Comment (Xottab-DUTY) [inline src/xrScriptEngine/script_engine.cpp]: Please, remove formatting changes

File: Externals/libtheora_static.vcxproj (hunk 1 above)
Comment (ChugunovRoman) [reply Externals/libtheora_static.vcxproj]: @Xottab-DUTY fixed

File: src/xrScriptEngine/script_engine.cpp (hunk 2 above)
Comment (ChugunovRoman) [reply src/xrScriptEngine/script_engine.cpp]: @Xottab-DUTY fixed

File: src/xray_input.props [hunk 3]
```diff
@@ -9,6 +9,7 @@
         $(xrSdkDir)include;
         $(xrExternals);
         $(xrExternals)luajit\src;
+        $(xrExternals)xrLuaFS\src;
```
Comment (Xottab-DUTY) [inline src/xray_input.props]:
```suggestion
```

File: src/xrScriptEngine/xrScriptEngine.vcxproj [hunk 4]
```diff
@@ -23,6 +23,11 @@
     <ClCompile>
       <PrecompiledHeaderFile>pch.hpp</PrecompiledHeaderFile>
       <PreprocessorDefinitions>_USRDLL;XRSCRIPTENGINE_EXPORTS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <AdditionalIncludeDirectories>
+        $(xrExternals)xrLuaFS;
```
Comment (Xottab-DUTY) [inline src/xrScriptEngine/xrScriptEngine.vcxproj]:
```suggestion
```

**Pull request closed without merging**